sdk/go/common/resource/plugin/host.go

```diff
@@ -88,7 +91,42 @@ type Host interface {
 
 // NewDefaultHost implements the standard plugin logic, using the standard installation root to find them.
 func NewDefaultHost(ctx *Context, config ConfigSource, runtimeOptions map[string]interface{},
-	disableProviderPreview bool) (Host, error) {
```

Comment:
actually given the only thing this is pulling from project is the providers map I think it would better to just pass that map as the parameter, decouples this from Projects and makes it a bit easier to set in tests.

Reply:
ProjectPlugins struct with Languages/Providers/Analyzers?
Might even make sense for the yaml to look like that as well?
```
plugins:
  providers:
  - name: aws
    path: ../../bin
```
?